backend/src/routes/preferences.js

```diff
@@ -125,13 +132,15 @@ router.get('/:userId/recommendations/listings', async (req, res, next) => {
 
             // TODO: This REQUIRES optimization for further milestones - too transactionally-heavy
             let rankedListings = [];
-            for (const id of generateListingRecommendations(scores)) {
+            for (const id of generateRecommendations(scores)) {
                 const currListing = await Listing.findById(id).lean();
                 rankedListings.push(currListing);
             }
             res.status(200).json(rankedListings);
         } else {
-            res.status(404).json({ error: 'No matching listings available!' });
+            let error = 'No matching listings available!';
+            res.status(404).json({ error: error });
+            next(error);
```

Comment:
I think this one is missing the `new Error(error)`

Reply:
😩 yes it is